scripts/lib/CIME/test_status.py

```diff
@@ -271,7 +287,54 @@ def _parse_test_status_file(self):
         with open(self._filename, "r") as fd:
             self._parse_test_status(fd.read())
 
-    def get_overall_test_status(self, wait_for_run=False, check_throughput=False, check_memory=False, ignore_namelists=False, ignore_memleak=False):
+    def _get_overall_status_based_on_phases(self, phases, wait_for_run=False, check_throughput=False, check_memory=False, ignore_namelists=False, ignore_memleak=False, no_run=False):
+
+        rv = TEST_PASS_STATUS
+        run_phase_found = False
+        for phase in phases: # ensure correct order of processing phases
+            if phase in self._phase_statuses:
```

Comment:
e3sm was having problems with incorrect test status reports for tests, so I refactored this code a bit. Basically, the idea is to give priority to the "core" phases when trying to determine the overall test status.